### PR TITLE
Add error message when "change" is first executed

### PR DESCRIPTION
**Pull request by XueqiangWei**
Signed-off-by: Xueqiang Wei <xuwei@redhat.com>

### CODE REVIEW

**Comment by XueqiangWei** [comment]
something new changed in rhel7.3:
Below error message will print in qmp when 'change' is first executed:
{"id": "Kl8WJXdJ", "error": {"class": "GenericError", "desc": "Tray of device 'drive-cd-disk' is not open"}}

In rhel7.2, error message like this:
{"id": "Kl8WJXdJ", "error": {"class": "GenericError", "desc": "Device 'drive-cd-disk' is locked"}}

**Comment by XueqiangWei** [comment]
ID 1332794

**Comment by XueqiangWei** [comment]
Results on RHEL7.2/RHEL7.3:
smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.change_media: PASS (134.53 s)
smp_16.16384m.repeat1.run_test.Host_RHEL.m7.u2.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.change_media: PASS (128.42 s)

File: qemu/tests/change_media.py [hunk 1]
```diff
@@ -116,9 +116,9 @@ def change_block(cmd=None):
     change_insert_cmd = "change device=%s,target=%s" % (device_name,
                                                         new_img_name)
     output = change_block(change_insert_cmd)
-    if "is locked" not in output:
-        msg = "%s is not locked after execute command '%s'" % (device_name,
-                                                               change_insert_cmd)
+    if "is locked" not in output and "is not open" not in output:
```
Comment (xutian) [inline qemu/tests/change_media.py]: if not ( 'is locked' in output or 'not open' in output):
         raise error

right?

Comment (XueqiangWei) [inline qemu/tests/change_media.py]: yes, it is. I will update, and I have sent email to developer to confirm if the return error message is ok.

**Comment by xutian** [comment]
looks ok to me.

File: qemu/tests/change_media.py [hunk 2]
```diff
@@ -116,9 +116,9 @@ def change_block(cmd=None):
     change_insert_cmd = "change device=%s,target=%s" % (device_name,
                                                         new_img_name)
     output = change_block(change_insert_cmd)
-    if "is locked" not in output:
-        msg = "%s is not locked after execute command '%s'" % (device_name,
-                                                               change_insert_cmd)
+    if not ("is locked" in output or "is not open" in output):
+        msg = "%s is not locked or is open after execute command '%s'" % (
+            device_name, change_insert_cmd)
```
Comment (PandaWei) [inline qemu/tests/change_media.py]: IMO, Here print the 'output' is more reason and precise for user to understand the reason.

**Comment by suqinhuang** [comment]
Acked-by: Suqin Huang

File: qemu/tests/change_media.py [hunk 3]
```diff
@@ -116,9 +116,10 @@ def change_block(cmd=None):
     change_insert_cmd = "change device=%s,target=%s" % (device_name,
                                                         new_img_name)
     output = change_block(change_insert_cmd)
-    if "is locked" not in output:
-        msg = "%s is not locked after execute command '%s'" % (device_name,
-                                                               change_insert_cmd)
+    if not ("is locked" in output or "is not open" in output):
+        msg = "%s is not locked or is open after execute command '%s'\n" % (
```
Comment (kongove) [inline qemu/tests/change_media.py]: Trivial syntax mistake?

%s is not locked or open after executed command: '%s'

**Comment by PyLearner** [comment]
Acked-by: Meng Yang  meyang@redhat.com

File: qemu/tests/change_media.py (hunk 3 above)
Comment (xutian) [inline qemu/tests/change_media.py]: can we make it looks more beautiful, like below:
```
msg = ("%s is not locked or is open "                               
             "after execute command %s "              
             "command output: %s " % (                
              device_name, change_insert_cmd, output)) 
```

Comment (XueqiangWei) [inline qemu/tests/change_media.py]: @kongove 
Trivial syntax mistake?
%s is not locked or open after executed command: '%s'

It is not syntax mistake, I just want to separate command from other words.

**Comment by xutian** [comment]
Looks ok to me, thanks @XueqiangWei  @PandaWei and @kongove .

**Pull request merged**